### PR TITLE
DAOS-8207 duns: update DUNS to be more inline with recent pool and cont API changes

### PR DESCRIPTION
**Pull request by mchaarawi**
- deprecate usage of uuids in the duns struct and shift usage to the string
  for cont and pool which can be used regardless whether this is a uuid or
  label in the connect/open APIs
- add a placeholder for system name, that is just set to NULL for now (not stored in the UNS)
- remove the recently added duns_set_pool/cont_label since those are just on the stack now in the struct and not required to be freed since they are always returned / required
- disable HDF5 VOL tests since there is an ABI change

(note that there is no need for a version bump since we already did that when we introduced the earlier changes in the UNS, we are reverting some of the new APIs that were added, so the version is already bumped from an earlier PR for users to check against)

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

**Review by johannlombardi — DISMISSED** [review]
We should really land this one for 2.0.

**Comment by mchaarawi** [comment]
> We should really land this one for 2.0.

yea i know, but im still not totally comfortable with this change so don't land it yet please.

File: src/client/dfs/duns.c
Comment (daltonbohning) [inline src/client/dfs/duns.c:709]: Do we have a standard or enforce formatted errors? In general - beyond this PR - I see different formats/styles. Just wondering if I **should** be leaving comments like this, or if it's not neessary
```suggestion
		D_ERROR("Failed to cleanup created container %s "DF_RC"\n", attrp->da_cont, DP_RC(rc2));
```

File: src/client/dfuse/ops/lookup.c
Comment (daltonbohning) [inline src/client/dfuse/ops/lookup.c:162]: Is this a TODO? (future)

File: src/utils/daos_hdlr.c
Comment (daltonbohning) [inline src/utils/daos_hdlr.c:216]: Was this one not meant to use the label / %s instead?

Comment (mchaarawi) [reply src/utils/daos_hdlr.c:216]: yea, but i think this is just dead code since it's only used from daos_old tool.. anyway i will update

Comment (mchaarawi) [reply src/utils/daos_hdlr.c:216]: actually this is a bigger change than i realized as there are many usage of this throughout, mostly with the old tool. iI will check those in a different PR. ill push some of the other fixes now here though.

File: src/client/dfs/duns.c
Comment (daltonbohning) [reply src/client/dfs/duns.c]: Definitely agree. That really was a genuine question. I think if we DO want to enforce this in the codebase then maybe it's time for a mass scrub...

File: src/utils/daos_hdlr.c
Comment (daltonbohning) [reply src/utils/daos_hdlr.c:216]: Fair enough. FWIW, this file has lots of inconsistencies with print statements :)

**Review by daltonbohning — DISMISSED** [review]

File: src/utils/daos_hdlr.c
Comment (mchaarawi) [reply src/utils/daos_hdlr.c:216]: indeed, and it was not my intention to fix all of it. i tried at least to fix the container uuid printing when using the daos tool. the pool ones seem mostly used from the daos_old tool since the pool handlers are implemented in the go side, and should be removed all together once we remove the daos.c (daos_old) tool

Comment (ashleypittman) [reply src/utils/daos_hdlr.c:216]: DH_PERROR_SYS() and DH_PERROR_DER() are intended to resolve the inconsistencies in this file, but it's a lot of effort to update them all and as stated some of it appears dead code.  As long as we're moving in the right direction then I think that's good enough.  Once it's stabilized and new features have landed/code is removed then we can do a pass just to migrate things over but right now unless the output is changing anyway it's just adding noise/conflicts.

Comment (daltonbohning) [reply src/utils/daos_hdlr.c:216]: Sure, makes sense!

File: src/client/dfs/duns.c
Comment (ashleypittman) [reply src/client/dfs/duns.c]: No we don't have a standard, NLT fault injection checking checks the use of DF_RC but not really much more, although it does at least attempt to ensure that common client error paths are at least not broken.  We also only run it on a tiny proportion of the client codebase so far.

File: src/client/dfs/duns.c
Comment (ashleypittman) [reply src/client/dfs/duns.c:682]: This came up in the other PR, NLT is only checking for "%d", errno for system error codes which is clearly worse than this, I guess we just don't run the fault injection tests over much code that uses system error codes yet.  The DH_PERROR_SYS() macro is for stderr of the "daos" binary, not log files but it seems like we might as well pick the same format for D_ERROR().

It's an NLT issue that it's only happy with the worst style of output, I'll try updating NLT to require some combination of strerror and the value and see how many places we need to update to get it to pass.

**Review by daltonbohning — DISMISSED** [review]

File: src/client/dfs/duns.c
Comment (mchaarawi) [reply src/client/dfs/duns.c:860]: good catch

File: src/client/dfs/duns.c
Comment (mchaarawi) [reply src/client/dfs/duns.c:646]: makes sense. but let me take this in a different PR when i add a dfs_cont_create_with_label so i can do it for both dfs and generic type containers rather than going and setting the property manually here.

File: src/client/dfs/duns.c
Comment (mchaarawi) [reply src/client/dfs/duns.c]: fixed

**Review by kccain — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**
